### PR TITLE
[optin/puller] task: update completed database upon payout tx finaliz…

### PR DESCRIPTION
**Pull request by Jaguar0625**
…ation

### CODE REVIEW

File: optin/puller/tests/client/test_SymbolClient.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+async def test_can_query_transaction_statuses(server):  # pylint: disable=redefined-outer-name
+	# Arrange:
+	client = SymbolClient(server.make_url(''))
+
+	# Act:
+	transaction_statuses = await client.transaction_statuses([HASHES[0], HASHES[2], HASHES[1]])
+
+	# Assert:
+	assert [f'{server.make_url("")}/transactionStatus'] == server.mock.urls
+	assert 3 == len(transaction_statuses)
```
Comment (gimre-xymcity) [inline optin/puller/tests/client/test_SymbolClient.py:106]: hmm, remind me, IRL, can this return less entries than queried hashes?

File: optin/puller/workflows/payout.py [hunk 2]
```diff
@@ -14,48 +16,74 @@
...
+	for address, request_group in unprocessed_requests.items():
+		if request_group.is_error:
+			# inconsistent error detection - what to do?
+			continue
+
+		if address in sent_requests or databases.completed.is_opted_in(address):
+			databases.inprogress.set_request_status(address, OptinRequestStatus.DUPLICATE, None)
+		else:
+			if databases.multisig.check_cosigners(address, [request.address for request in request_group.requests]):
+				sent_transaction_hash = send_funds(request_group.requests[0].destination_public_key)
```
Comment (gimre-xymcity) [inline optin/puller/workflows/payout.py:60]: one thing is not clear to me (bit unrelated to this review), where do we send _non_ multisig payouts?

File: optin/puller/tests/client/test_SymbolClient.py (hunk 1 above)
Comment (Jaguar0625) [reply optin/puller/tests/client/test_SymbolClient.py:106]: yes, if the hash is in none of the DB collections

File: optin/puller/workflows/payout.py (hunk 2 above)
Comment (Jaguar0625) [reply optin/puller/workflows/payout.py:60]: here ^^

File: optin/puller/workflows/payout.py [hunk 3]
```diff
@@ -14,48 +16,74 @@
...
 	# TODO_: implement this
 	return Hash256('')
 
 
+class Databases:
+	def __init__(self, completed_connection, inprogress_connection, multisig_connection):
+		self.completed = CompletedOptinDatabase(completed_connection)
+		self.inprogress = InProgressOptinDatabase(inprogress_connection)
+		self.multisig = MultisigDatabase(multisig_connection)
+
```
Comment (gimre-xymcity) [inline optin/puller/workflows/payout.py:32]: plug in balances db?

**Review by gimre-xymcity — DISMISSED** [review]

File: optin/puller/workflows/payout.py [hunk 4]
```diff
@@ -14,48 +17,101 @@
...
 
 
 def send_funds(_):
 	# TODO_: implement this
 	return Hash256('')
 
 
+# region Databases
+
+class Databases:
```
Comment (gimre-xymcity) [inline optin/puller/workflows/payout.py:30]: do we want to move this somewhere _outside?_

**Review by gimre-xymcity — DISMISSED** [review]

**Review by gimre-xymcity — DISMISSED** [review]

**Pull request merged**